src/outpack/location.py

```diff
@@ -71,6 +82,149 @@ def outpack_location_rename(old, new, root=None, *, locate=True):
     update_config(config, root.path)
 
 
+def location_resolve_valid(
+    location, root, include_local, include_orphan, allow_no_locations
+):
+    if location is None:
+        location = outpack_location_list(root)
+    elif isinstance(location, str):
+        if location not in outpack_location_list(root):
+            msg = f"Unknown location: '{location}'"
+            raise Exception(msg)
+        else:
+            location = [location]
+    elif isinstance(location, collections.abc.Iterable) and all(
+        isinstance(item, str) for item in location
+    ):
+        unknown = set(location).difference(outpack_location_list(root))
+        if len(unknown) > 0:
+            unknown_text = "', '".join(unknown)
+            msg = f"Unknown location: '{unknown_text}'"
+            raise Exception(msg)
+    else:
+        msg = (
+            "Invalid input for 'location'; expected None or a list of "
+            "strings"
+        )
+        raise Exception(msg)
+
+    if not include_local and LOCATION_LOCAL in location:
+        location.remove(LOCATION_LOCAL)
+    if not include_orphan and LOCATION_ORPHAN in location:  # pragma: no cover
+        location.remove(LOCATION_ORPHAN)
```

Comment:
```suggestion
    if not include_local:
        location.remove(LOCATION_LOCAL)
    if not include_orphan:  # pragma: no cover
        location.remove(LOCATION_ORPHAN)
```

Reply:
Actually `location.remove(x)` throws an error if `x` is not in the list